sw/device/boot_rom/rom_link.ld

```diff
@@ -100,9 +104,19 @@ SECTIONS {
    */
   .data ORIGIN(ram): AT(_data_init_start) ALIGN(4) {
     _data_start = .;
+    __global_pointer$ = . + 2048;
+
+    /* Small data should come before larger data. This helps to ensure small
+     * globals are within 2048 bytes of the value of `gp`, making their accesses
+     * hopefully only take one instruction. */
+    *(.sdata)
+    *(.sdata.*)
```

Comment:
Is there a reason we don't have a layout like the following?
```
.sdata
__global_pointer$
.sbss
.data
.bss
```
While this means more CRT code, it increases the chance of not having to jump over .data to get to .sbss.

Reply:
I would prefer to keep the zero-initialised parts of the binary contiguous, to make the CRT simpler, at the moment. (This also applies to `.idata`, which I'd prefer to be copied to `.data` in one contiguous pass).

I'd be happy to revisit this choice further down the line once our firmware is quite a bit larger and this makes more of a difference.

Reply:
SGTM